legate/core/shape.py

```diff
@@ -14,38 +14,48 @@
 #
 from __future__ import annotations
 
-from collections.abc import Iterable
 from functools import reduce
+from typing import TYPE_CHECKING, Iterable, Iterator, Optional, Union
 
+from typing_extensions import TypeAlias, overload
 
-def _cast_tuple(value, ndim):
+if TYPE_CHECKING:
+    from . import IndexSpace
+    from .runtime import Runtime
+
+Extentable: TypeAlias = Union["Shape", int, Iterable[int]]
+
+
+def _cast_tuple(value: Extentable, ndim: int) -> tuple[int, ...]:
     if isinstance(value, Shape):
         return value.extents
-    elif isinstance(value, tuple):
-        return value
+    elif isinstance(value, Iterable):
+        return tuple(value)
```

Comment:
With these changes, `_cast_tuple` can be used in place of duplicated code in `Shape.__init__`

Reply:
That's great.